### PR TITLE
Docker semantics

### PR DESCRIPTION
**Pull request by jsok**
A docker image is a fully qualified string of the form:

`<registry>/<repository>:<tag>`

The semantics in `ci2` are fuzzy/wrong in this department.

This PR renames to accurately reflect their actual content:

 1. `DOCKER_IMAGE` to `DOCKER_REPO`
 1. `DOCKER_TAG_NAME` to `DOCKER_IMAGE`

I suspect this is a sufficiently breaking change to merit a major version bump on the project.

### CODE REVIEW

File: src/main/bash/docker-support.sh [hunk 1]
```diff
@@ -48,22 +48,22 @@ function usage() {
...
-  echo "  -s                    - publish the site also"
-  echo "  -e <CI_EMAIL>        - used for _site doc publish-ing - defaults to ${CI_EMAIL}" 1>&2
-  echo "  -n <CI_USERNAME>     - used for _site doc publish-ing - defaults to ${CI_USERNAME}" 1>&2
+  echo "publish -e <git-email> -n <git-username> [ -i <docker-repo> | -d <docker-image> ] [ -s ]"
+  echo "  -s                     - publish the site also"
+  echo "  -e <CI_EMAIL>          - used for _site doc publish-ing - defaults to ${CI_EMAIL}" 1>&2
+  echo "  -n <CI_USERNAME>       - used for _site doc publish-ing - defaults to ${CI_USERNAME}" 1>&2
   echo
-  echo "  -i <DOCKER_IMAGE>     - Docker tag .. user/tag" 1>&2
+  echo "  -i <DOCKER_REPO>       - Docker repository, e.g. user/tag" 1>&2
```
Comment (rbuckland) [inline src/main/bash/docker-support.sh]: Agree on the change (the old was wrong) .. but are these correct ? 
DOCKER_REPO <-- wouldn't that be docker.some.internal.host

File: src/main/bash/docker-support.sh [hunk 2]
```diff
@@ -48,22 +48,22 @@ function usage() {
...
   echo " OR" 1>&2
-  echo "  -d <DOCKER_TAG_NAME>  - Docker tag .. registry/user/tag" 1>&2
+  echo "  -d <DOCKER_IMAGE>      - Docker image, e.g. registry/repo:tag" 1>&2
```
Comment (rbuckland) [inline src/main/bash/docker-support.sh]: similar comment, that "comment" example doesn't seem to match the ENV VAR name

File: src/test/bash/run-tests.sh [hunk 3]
```diff
@@ -46,5 +51,5 @@ function run_tests() {
 # main()
 #
 init
-run_tests
+run_tests "$cases"
```
Comment (rbuckland) [inline src/test/bash/run-tests.sh]: simple change - good.

File: src/main/bash/docker-support.sh (hunk 1 above)
Comment (jsok) [inline src/main/bash/docker-support.sh]: No, a repo is simply the path of a registry. I can push the same repo `foo/bar` to any registry `docker.io`, `quay.io`, etc.

Comment (jsok) [inline src/main/bash/docker-support.sh]:
```
docker push --help

Usage:  docker push [OPTIONS] NAME[:TAG]

Push an image or a repository to a registry
```

**Comment by jsok** [comment]
Also see https://docs.docker.com/registry/introduction/#understanding-image-naming

**Comment by stephanh** [comment]
@rbuckland @jsok do we want to resurrect this PR again?

**Pull request closed without merging**